src/DIRAC/Resources/Cloud/cloudinit.template

```diff
@@ -0,0 +1,183 @@
+       chmod +x dirac-install.py 
+       for SRC in dirac-pilot.py pilotCommands.py pilotTools.py PilotLogger.py PilotLoggerTools.py MessageSender.py; do
+         curl -o ${SRC} https://raw.githubusercontent.com/DIRACGrid/Pilot/master/Pilot/${SRC}
+       done
+     fi
+     if [ -d /cvmfs/grid.cern.ch/etc/grid-security/certificates ]; then
+       curl -o pilot.json --capath /cvmfs/grid.cern.ch/etc/grid-security/certificates \
+               https://%(pilot-server)s/pilot/pilot.json
+     else
+       curl -o pilot.json -k https://%(pilot-server)s/pilot/pilot.json
+     fi
+     python dirac-pilot.py \
+            --setup %(setup)s \
+            -r %(release-version)s \
+            --MaxCycles 10 \
+            --configurationServer %(cs-servers)s \
+            --Name "%(ce-name)s" \
+            --name "%(dirac-site)s" \
+            -Q "%(image-name)s" \
+            --cert \
+            -o /LocalSite/VMID=%(vm-uuid)s \
+            -o /LocalSite/LocalCE=Singularity \
+            -o /Resources/Computing/CEDefaults/VirtualOrganization=%(vo)s \
+            -o /Resources/Computing/CEDefaults/WholeNode=%(whole-node)s \
+            -o /Resources/Computing/CEDefaults/ContainerExtraOpts="%(extraopts)s" \
+            -o /Resources/Computing/CEDefaults/ContainerRoot=%(user-root)s
+ - path: /root/run_monitor.sh
+   permissions: '0755'
+   content: |
+     #!/bin/bash
+     mkdir -p /mnt/monitor
+     exec >/mnt/monitor/monitor_start.log 2>&1
+     cd /mnt/monitor
+     if [ -f /cvmfs/dirac.egi.eu/pilot/pilot.tar ]; then
+       tar xf /cvmfs/dirac.egi.eu/pilot/pilot.tar dirac-install.py
+     else
+       curl -o dirac-install.py https://raw.githubusercontent.com/DIRACGrid/DIRAC/integration/Core/scripts/dirac-install.py
+       chmod +x dirac-install.py
+     fi
+
+     python dirac-install.py -r %(release-version)s
+     mkdir -p etc/grid-security
+     cp /root/hostkey.pem etc/grid-security/hostkey.pem
+     chmod 600 etc/grid-security/hostkey.pem
+     ln -s hostkey.pem etc/grid-security/hostcert.pem
+     source bashrc
+     dirac-configure -S %(setup)s \
+                     -C %(cs-servers)s \
+                     -o /Resources/Computing/CEDefaults/VirtualOrganization=%(vo)s \
+                     -o /DIRAC/Security/CertFile=/mnt/monitor/etc/grid-security/hostcert.pem \
+                     -o /DIRAC/Security/KeyFile=/mnt/monitor/etc/grid-security/hostkey.pem \
+                     -o /LocalSite/RunningPod=%(running-pod)s \
+                     -o /LocalSite/VMID=%(vm-uuid)s \
+                     -o /Cloud/%(running-pod)s/JobWrappersLocation=/mnt/dirac \
+                     --UseServerCertificate
+     mkdir -p runit/VirtualMachineMonitorAgent/log
+     exec python /mnt/monitor/DIRAC/Core/scripts/dirac-agent.py WorkloadManagement/VirtualMachineMonitorAgent >runit/VirtualMachineMonitorAgent/log/current 2>&1 
```

Comment:
Installing this agent is a bit awkward... It runs as the root user, so is installed completely independently to the main pilot which runs as "dirac" in the VM (hence the dirac-install reference). I think the agent was originally designed to be built-in to the images, but we prefer to support running from "plain" images to make things simpler.

Reply:
What is the reason for running as root?

Reply:
This idea is that by running as root it can tidy up and eventually halt the instance if needed, although this hasn't actually been properly implemented in the current version.